src/webview/VSCodeKaotoEditorChannelApi.ts

```diff
@@ -63,4 +63,20 @@ export class VSCodeKaotoEditorChannelApi extends DefaultVsCodeKieEditorChannelAp
   async saveResourceContent(relativePath: string, content: string): Promise<void> {
     vscode.window.showErrorMessage(`saveResourceContent Not implemented. It was called with ${relativePath} and ${content}`);
   }
+
+  async askUserForFileSelection(include: string, exclude?: string, options?: Record<string, unknown>): Promise<string[] | string | undefined> {
+    try {
+      const workspaceFolder = vscode.workspace.getWorkspaceFolder(this.currentEditedDocument.uri);
+      if (!workspaceFolder) return;
+      const includePattern = new vscode.RelativePattern(workspaceFolder, include);
+      const files = await vscode.workspace.findFiles(includePattern, exclude);
+      return await vscode.window.showQuickPick(files.map((f) => {
+        return path.relative(workspaceFolder.uri.fsPath, f.path);
```

Comment:
If the `include` filter is `**/*.xsd` for example, `vscode.workspace.findFiles()` also returns the files those are in sub directories.

Reply:
ah now I got what is the problem, so `getResourceContent` is resolving a file through a relative path "from Camel YAML file", not the workspace root, so this method also have to align with that... hmm so currently we have to use the same directory which contains the runtime files... (the primary use of this method is to resolve design time only schema files from workspace)

Reply:
Maybe split or add a switch flag to `getResourceContent` to choose whether the relative path is from Camel YAML file or workspace?

Reply:
We can never know what will be the workspace folder. it can be directly the parent, it can be some parent folders, it can be some Maven multimodule with several layer of hierarchy and so several parent folders. For the same project, it can be opened from different level too

Reply:
What do you suggest then? Is the "relative path from KaotoMetadataFile" any better?

Reply:
the xsd file is metadata for a specific Camel file. I think to have the path relative to the Camel file is the more robust way. (after having the metadata directly in the camel file but this possibility was discarded)

Reply:
We talked a bit further and concluded that for now, we use "relative path from Camel YAML file" for the path, but search files from ancestor directories by tracing up to the topmost directory in the workspace.

Reply:
Done - for now it searches the file only from the associated workspace folder, and use the relative path from the Camel YAML file currently associated with the Editor. Since using other workspace folders than the one associated with the Camel YAML file could introduce a portability issue, it would need more thought.